packages/data-provider/test/utils/submission/parseRecordsToInsert.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+
+import type { SubmissionInsertData } from '@overture-stack/lyric-data-model/models';
+
+import { parseRecordsToInsert } from '../../../src/utils/recordsParser.js';
+import type { EntityData, SchemasDictionary } from '../../../src/utils/types.js';
+
+describe('parseRecordsToInsert', () => {
+	const inventoryDictionary: SchemasDictionary = {
+		name: 'inventory',
+		version: '1.0.0',
+		schemas: [
+			{
+				name: 'user',
+				fields: [
+					{ name: 'id', valueType: 'number' },
+					{ name: 'name', valueType: 'string' },
+					{ name: 'birthYear', valueType: 'integer' },
+					{ name: 'hasAllergies', valueType: 'boolean' },
+				],
+			},
+			{
+				name: 'product',
+				fields: [
+					{ name: 'id', valueType: 'number' },
+					{ name: 'name', valueType: 'string' },
+				],
+			},
+		],
+	};
+	it('should build insert records correctly for valid input', () => {
```

Comment:
This statement is unclear. If this failed I would not know what the test is attempting to do and likely wouldn't be able to fix it.